dbt_platform_helper/domain/codebase.py

```diff
@@ -0,0 +1,298 @@
+from dbt_platform_helper.utils.application import ApplicationNotFoundError
+from dbt_platform_helper.utils.application import load_application
+from dbt_platform_helper.utils.aws import get_aws_session_or_abort
+from dbt_platform_helper.utils.files import mkfile
+from dbt_platform_helper.utils.template import setup_templates
+
+
+class Codebase:
+    def __init__(
+        self,
+        input_fn: Callable[[str], str] = click.prompt,
+        echo_fn: Callable[[str], str] = click.secho,
+        confirm_fn: Callable[[str], bool] = click.confirm,
+        load_application_fn: Callable[[str], Application] = load_application,
+        get_aws_session_or_abort_fn: Callable[[str], Session] = get_aws_session_or_abort,
+    ):
+        self.input_fn = input_fn
+        self.echo_fn = echo_fn
+        self.confirm_fn = confirm_fn
+        self.load_application_fn = load_application_fn
+        self.get_aws_session_or_abort_fn = get_aws_session_or_abort_fn
+
+    def prepare(self):
+        """Sets up an application codebase for use within a DBT platform
+        project."""
+        templates = setup_templates()
+
+        repository = (
+            subprocess.run(["git", "remote", "get-url", "origin"], capture_output=True, text=True)
+            .stdout.split("/")[-1]
+            .strip()
+            .removesuffix(".git")
+        )
+
+        if repository.endswith("-deploy") or Path("./copilot").exists():
+            self.echo_fn(
+                "You are in the deploy repository; make sure you are in the application codebase repository.",
+                fg="red",
+            )
+            exit(1)
+
+        builder_configuration_url = "https://raw.githubusercontent.com/uktrade/ci-image-builder/main/image_builder/configuration/builder_configuration.yml"
+        builder_configuration_response = requests.get(builder_configuration_url)
+        builder_configuration_content = yaml.safe_load(
+            builder_configuration_response.content.decode("utf-8")
+        )
+        builder_versions = next(
+            (
+                item
+                for item in builder_configuration_content["builders"]
+                if item["name"] == "paketobuildpacks/builder-jammy-base"
+            ),
+            None,
+        )
+        builder_version = max(x["version"] for x in builder_versions["versions"])
+        # Temporary hack until https://uktrade.atlassian.net/browse/DBTP-351 is done
+        # Will need a change in tests/platform_helper/expected_files/.copilot/config.yml, when removed.
+        builder_version = min(builder_version, "0.4.240")
+
+        Path("./.copilot/phases").mkdir(parents=True, exist_ok=True)
+        image_build_run_contents = templates.get_template(f".copilot/image_build_run.sh").render()
+
+        config_contents = templates.get_template(f".copilot/config.yml").render(
+            repository=repository, builder_version=builder_version
+        )
+
+        self.echo_fn(
```

Comment:
Thinking about if we ever had click (echo) functionality at command level, would these domain class commands require split up between echo's to support "echoing" progress

Reply:
sorry i dont get this can you explain this a bit?

Reply:
sorry, we can ignore my comment. 
What I was thinking at the time was that the `click` functionality should be at the command layer but as you call it here would you need to split the function up if you were to move the echo to the command layer. 
Although this does not matter as techinically `self.echo_fn` could be any logging function it could be strout but also anything else too.